File: Programmers/Lv. 2/H-Index.py
# def solution(citations):
#     for i in range(max(citations), -1, -1):
#         if len(list(filter(lambda x : x >= i, citations))) >= i:
#             return i
        
def solution(citations):
    citations.sort(reverse=True)
    answer = max(map(min, enumerate(citations, start=1)))
    return answer

File: Programmers/Lv. 2/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int max = *max_element(citations.begin(), citations.end());
    for (int i = max; i > 0; i--)
    {
        int cnt = 0;
        for (int c : citations)     
        {
            if (c >= i)
            {
                cnt++;
            }
        }
        if (cnt >= i)
        {
            return i;
        }
    }
}